seq2seq/dataset/utils.py

```diff
@@ -55,13 +55,13 @@ def prepare_data(path, src_max_len, tgt_max_len, tokenize_func=space_tokenize):
                 if filter_pair(pair, src_max_len, tgt_max_len):
                     pairs.append(pair)
             except:
-                print("Error when reading line: {0}".format(line))
+                logger.error("Error when reading line: {0}".format(line))
                 raise
             counter += 1
             if counter % 100 == 0:
-                print("\rRead {0} lines".format(counter), end="")
+                logger.info("Read {0} lines".format(counter))
```

Comment:
I'm being demanding here...  It wasn't a problem when using printing because we could use '\r'.  But now when processing a big dataset, they would produce too many lines.  For example, WMT data has 4.5 million pairs, that's 45k lines of logging!

I'm thinking we can go over the dataset twice: 
1. The first time just to count the number of lines, which would cost about 1 second for 4.5m lines
2. For the second time, we do the pre-processing and log the percentage process for every 5% or 10%.

Same thing can be done for line 98.  So, better do it with a function.